### PR TITLE
HBW-183 Improve claim button styles on widget form

### PR DESCRIPTION
**Pull request by mxsmirnoff**

### CODE REVIEW

**Comment by mxsmirnoff** [comment]
image

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ v1.8.0 [unreleased]
 - [#347](https://github.com/latera/homs/pull/347) Improve claim buttons styles.
 - [#349](https://github.com/latera/homs/pull/349) Change colors of priorities.
 - [#346](https://github.com/latera/homs/pull/346) Add code coverage for backend.
+- [#351](https://github.com/latera/homs/pull/351) Improve claim button styles on widget form and add test for claiming task.
```
Comment (TimAle) [inline CHANGELOG.md]: Not sure that info about test should be placed to changelog.

File: hbw/app/javascript/packs/hbw/components/form.js.jsx [hunk 2]
```diff
@@ -84,7 +75,8 @@ modulejs.define('HBWForm', ['React', 'jQuery', 'HBWError', 'HBWFormDatetime',
         value:           this.props.variables[name],
         formSubmitting:  this.state.submitting || this.state.fileUploading,
         env:             this.props.env,
-        fileListPresent: this.fileListPresent(this.props.form.fields)
+        fileListPresent: this.fileListPresent(this.props.form.fields),
+        hideSubmit:      !this.props.assignee
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/form.js.jsx]: showSubmit

File: spec/hbw/features/task_claiming_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  describe 'form with all fields' do
+    before(:each) do
+      click_on 'Orders'
+      expect(page).to have_content 'Orders list'
+      expect_widget_presence
+    end
+
+    scenario 'have only disables or readonly fields' do
```
Comment (sugarfree1) [inline spec/hbw/features/task_claiming_spec.rb]: disabled

File: hbw/app/controllers/hbw/tasks_controller.rb [hunk 4]
```diff
@@ -18,6 +18,7 @@ def index
 
     def claiming
       @tasks = widget.claiming_task_list(current_user_identifier, entity_class, assigned, max_results, search_query)
+      head :no_content
```
Comment (TimAle) [inline hbw/app/controllers/hbw/tasks_controller.rb]: We return JSON with tasks, why `no_content`?

Comment (mxsmirnoff) [reply hbw/app/controllers/hbw/tasks_controller.rb]: Oops, wrong place :(

File: hbw/app/javascript/packs/hbw/components/form.js.jsx [hunk 5]
```diff
@@ -84,7 +77,8 @@ modulejs.define('HBWForm', ['React', 'jQuery', 'HBWError', 'HBWFormDatetime',
         value:           this.props.variables[name],
         formSubmitting:  this.state.submitting || this.state.fileUploading,
         env:             this.props.env,
-        fileListPresent: this.fileListPresent(this.props.form.fields)
+        fileListPresent: this.fileListPresent(this.props.form.fields),
+        showSubmit:      this.props.assignee
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/form.js.jsx]: add ```!!```

**Comment by mxsmirnoff** [comment]
Updated

**Pull request merged**